include/aligator/modelling/dynamics/continuous-centroidal-fwd.hpp

```diff
@@ -0,0 +1,69 @@
+/// @copyright Copyright (C) 2022 LAAS-CNRS, INRIA
+#pragma once
+
+#include "aligator/modelling/dynamics/ode-abstract.hpp"
+
+#include <proxsuite-nlp/modelling/spaces/vector-space.hpp>
+
+namespace aligator {
+namespace dynamics {
+template <typename Scalar> struct CentroidalFwdDataTpl;
+
+/**
+ * @brief   Nonlinear centroidal forward dynamics with smooth control.
+ *
+ * @details This is described in state-space \f$\mathcal{X} = T\mathcal{Q}\f$
+ * (the phase space \f$x = (c,h,L,u)\f$ with c CoM position, h linear momentum,
+ * L angular momentum and u forces) using the differential equation \f[ \dot{x}
+ * = f(x, u) = F_x * x + F_u(x) * u \f] as described by the Newton-Euler law
+ * of momentum. Control parameter is set to be the derivative of u to ensure
+ * force smoothness.
+ *
+ */
+template <typename _Scalar>
+struct ContinuousCentroidalFwdDynamicsTpl : ODEAbstractTpl<_Scalar> {
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+  using Scalar = _Scalar;
+  ALIGATOR_DYNAMIC_TYPEDEFS(Scalar);
+  using Base = ODEAbstractTpl<Scalar>;
+  using BaseData = ODEDataTpl<Scalar>;
+  using ContDataAbstract = ContinuousDynamicsDataTpl<Scalar>;
+  using Data = ContinuousCentroidalFwdDataTpl<Scalar>;
+  using Manifold = proxsuite::nlp::VectorSpaceTpl<Scalar>;
+  using ManifoldPtr = shared_ptr<Manifold>;
+  using Matrix3s = Eigen::Matrix<Scalar, 3, 3>;
+
+  using Base::nu_;
+
+  ManifoldPtr space_;
+  const std::size_t nk_;
+  const double mass_;
```

Comment:
No `const` data members
```suggestion
  std::size_t nk_;
  double mass_;
```